### PR TITLE
Fix oauth window popup issue

### LINKED ISSUE #4553: No oauth window appear after clicking the Google sign-in button
Description:
There is no OAuth window appearing when the user clicks on the Google sign-in button. I'm unable to find out the root cause of it for the moment.

From the user:
>  I just reinstalled Chrome, and I'm trying to set it up
again.
When I click the "Sign in with Google" button and expect to see the
oAuth2 popup window, I get the attached error message pop up instantly
instead. There is no oAuth window. It doesn't matter what Google
account I try to select in the upper right corner either. 

screenshot:
image

<details><summary>stack traces: </summary>

```
DevTools failed to load source map: Could not load content for chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/forge.js.map: System error: net::ERR_FILE_NOT_FOUND
DevTools failed to load source map: Could not load content for chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/bootstrap/bootstrap.min.js.map: System error: net::ERR_FILE_NOT_FOUND
DevTools failed to load source map: Could not load content for chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/bootstrap/bootstrap.min.css.map: System error: net::ERR_FILE_NOT_FOUND
catch.js:60 Error: failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
    at Catch.nameAndDetailsAsException (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:345:15)
    at Catch.report (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:98:34)
    at Settings.newGoogleAcctAuthPromptThenAlertOrForward (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/settings.js:361:23)
    at async SettingsView.<anonymous> (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/index.js:201:28)


details:
[typeof:object:[object Object]] {"result":"Error","error":"No oauth window returned after initiating it","acctEmail":"some_email@domain.test"}
Catch.onErrorInternalHandler @ catch.js:60
Catch.reportErr @ catch.js:92
Catch.report @ catch.js:98
Settings.newGoogleAcctAuthPromptThenAlertOrForward @ settings.js:361
await in Settings.newGoogleAcctAuthPromptThenAlertOrForward (async)
(anonymous) @ index.js:201
cbWithErrsHandled @ ui.js:74
(anonymous) @ ui.js:110
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
catch.js:64 failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
Error: failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
    at Catch.nameAndDetailsAsException (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:345:15)
    at Catch.report (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:98:34)
    at Settings.newGoogleAcctAuthPromptThenAlertOrForward (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/settings.js:361:23)
    at async SettingsView.<anonymous> (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/index.js:201:28)


details:
[typeof:object:[object Object]] {"result":"Error","error":"No oauth window returned after initiating it","acctEmail":"some_email@domain.test"}
Catch.onErrorInternalHandler @ catch.js:64
Catch.reportErr @ catch.js:92
Catch.report @ catch.js:98
Settings.newGoogleAcctAuthPromptThenAlertOrForward @ settings.js:361
await in Settings.newGoogleAcctAuthPromptThenAlertOrForward (async)
(anonymous) @ index.js:201
cbWithErrsHandled @ ui.js:74
(anonymous) @ ui.js:110
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
index.htm:1 Unchecked runtime.lastError: Invalid value for bounds. Bounds must be at least 50% within visible screen space.
catch.js:291 FlowCrypt ERROR: Please report errors above to human@flowcrypt.com. We fix errors VERY promptly.
```

</details>

Reference: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgzGpGwmbmBXStgffvgCpbjFPQQCV

### PR DESCRIPTION
**Pull request by martgil**
This PR resizes the OAuth window popup and will try to open the OAuth window on the active screen instead of the primary screen.

close #4553

----------------------------------

**Tests** _(delete all except exactly one)_:
- Difficult to test (need to emulate multiple screens in Chromium)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by martgil** [comment]
@rrrooommmaaa sir, here is the draft PR. Thank you!

File: extension/js/common/api/email-provider/gmail/google-auth.ts [hunk 1]
```diff
@@ -121,7 +121,11 @@ export class GoogleAuth {
     }
     const authRequest: AuthReq = { acctEmail, scopes, csrfToken: `csrf-${Api.randomFortyHexChars()}` };
     const url = GoogleAuth.apiGoogleAuthCodeUrl(authRequest);
-    const oauthWin = await windowsCreate({ url, left: 100, top: 50, height: 800, width: 550, type: 'popup' });
+    const adaptiveHeight = Math.floor((window.innerHeight || window.screen.height) * 0.9);
+    const adaptiveWidth = Math.floor((window.innerWidth || window.screen.width) * 0.4);
```
Comment (rrrooommmaaa) [inline extension/js/common/api/email-provider/gmail/google-auth.ts:125]: Thank you, @martgil.
I tested this in various multi-screen setups and it works ok, however, I think, we'd better calculate the adaptive width and height based on the screen boundaries (e.g. `const screenHeight = window.screen.height || window.innerHeight` etc.) as the popup window can be placed beyond the browser window (and it actually gets placed this way in certain conditions). This is how it looks now when there is a small browser window on a big screen.
image
We can also increase the adaptiveWidth up to `* 0.9` if it is less than the comfortable width (550?)
We can also extract the non-standard properties without using `any` via `unknown` conversion and please add an explanation
```
// non-standard but supported by most of the browsers
const { availLeft, availTop } = (window.screen as unknown as { availLeft: number, availTop: number });
```

Comment (rrrooommmaaa) [reply extension/js/common/api/email-provider/gmail/google-auth.ts:125]: Even better:
```
const { availLeft, availTop } = (window.screen as unknown as { availLeft?: number, availTop?: number });
```

File: extension/js/common/api/email-provider/gmail/google-auth.ts [hunk 2]
```diff
@@ -121,7 +121,11 @@ export class GoogleAuth {
...
+    const leftOffset = Math.floor((window.screen.width / 2) - (adaptiveWidth / 2) + (window as any).screen.availLeft); // tslint:disable-line:no-unsafe-any
+    const topOffset = Math.floor((window.innerHeight / 2) - (adaptiveHeight / 2) + (window as any).screen.availTop); // tslint:disable-line:no-unsafe-any
```
Comment (rrrooommmaaa) [inline extension/js/common/api/email-provider/gmail/google-auth.ts:127]: based on `screenHeight` value introduced above

File: extension/js/common/api/email-provider/gmail/google-auth.ts (hunk 1 above)
Comment (martgil) [reply extension/js/common/api/email-provider/gmail/google-auth.ts]: Thanks a lot for your help, sir! It now looks even better. I can now notice that prioritizing `window.innerHeight` creates a smaller OAuth window when the browser window is not in fullscreen mode (maximized). Code snippet for getting specific properties from `window.screen` + using unknown type is helpful -- I can use this later.

File: extension/js/common/api/email-provider/gmail/google-auth.ts (hunk 2 above)
Comment (martgil) [reply extension/js/common/api/email-provider/gmail/google-auth.ts]: Ok, sir. I tried using the screenHeight over window.innerHeight and it works nicely.

**Review by rrrooommmaaa — APPROVED** [review]

**Comment by ioanmo226** [comment]
It seems like this PR broken live gmail test.
https://flowcrypt.semaphoreci.com/jobs/866f120d-0c1d-4cda-ac01-c61325804d2d

**Pull request merged**